### PR TITLE
fix(ripple): check parent before calling removeChild

### LINKED ISSUE #13457: [Bug Report] VRipple - NotFoundError: The object can not be found here.
### Environment
**Vuetify Version:** 2.4.9
**Vue Version:** 2.6.11
**Browsers:** Safari (Mobile Safari UI/WKWebView 14.4)
**OS:** iOS

### Steps to reproduce
Unable to reproduce myself - this is an error from a user's device that was captured by our error logs. Not even sure what conditions reproduce this. However, we're seeing this error in the error logs for our production app, and it seems clear enough what's going on - https://github.com/vuetifyjs/vuetify/blob/ed6f621f9f9086382c2894ea695355ae0bdd1ab8/packages/vuetify/src/directives/ripple/index.ts#L151 is removing a node from a specific parent when that parent doesn't seem to own the node anymore.

### Expected Behavior
No errors

### Actual Behavior
```
Error: NotFoundError: The object can not be found here.
    window.onerror@webpack:///node_modules/vuetify/dist/vuetify.js:33440:47
```

This is the full stack trace - there is no more detail in the logs than this. This line maps to https://github.com/vuetifyjs/vuetify/blob/ed6f621f9f9086382c2894ea695355ae0bdd1ab8/packages/vuetify/src/directives/ripple/index.ts#L151

### Reproduction Link
No reproduction link, unfortunately. I think this should be fixable without a repro by just adding a check that the element being removed does belong to the parent it is being removed from before removing it.

### PR DESCRIPTION
**Pull request by qngl**
## Description
Check parent before calling removeChild in v-ripple ( #13457 )

fixes #13457

## Motivation and Context
User actions with ripple effect in iOS devices throw quite a few those errors.

## How Has This Been Tested?
Only 1 line non-breaking change, so no additional test created. 

## Markup:

<details>

```vue
```
</details>

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any features but makes things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and backwards compatible changes and `next` for non-backwards compatible changes).
- [x] My code follows the code style of this project.
- [ ] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)

### CODE REVIEW

**Review by MoWafa — APPROVED** [review]
LGTM, Thanks @qngl.

**Comment by shadow-light** [comment]
This fills my error logs, please merge :)

File: packages/vuetify/src/directives/ripple/index.ts [hunk 1]
```diff
@@ -139,7 +139,7 @@ const ripples = {
           delete el.dataset.previousPosition
         }
 
-        animation.parentNode && el.removeChild(animation.parentNode)
+        animation.parentNode && el === animation.parentNode.parentNode && el.removeChild(animation.parentNode)
```
Comment (johnleider) [inline packages/vuetify/src/directives/ripple/index.ts:142]: I haven't tested this, but could we just use optional chaining?

`if(animation?.parentNode?.parentNode === el) el.removeChild(animation.parentNode)`

**Review by johnleider — CHANGES_REQUESTED** [review]

**Comment by MoWafa** [comment]
Any update on this?!!

File: packages/vuetify/src/directives/ripple/index.ts (hunk 1 above)
Comment (KaelWD) [reply packages/vuetify/src/directives/ripple/index.ts:142]:
```suggestion
        if(animation?.parentNode?.parentNode === el) el.removeChild(animation.parentNode)
```

File: packages/vuetify/src/directives/ripple/index.ts [hunk 2]
```diff
@@ -139,7 +139,7 @@ const ripples = {
           delete el.dataset.previousPosition
         }
 
-        animation.parentNode && el.removeChild(animation.parentNode)
+        if(animation?.parentNode?.parentNode === el) el.removeChild(animation.parentNode)
```
Comment (KaelWD) [inline packages/vuetify/src/directives/ripple/index.ts:142]:
```suggestion
        if (animation?.parentNode?.parentNode === el) el.removeChild(animation.parentNode)
```

File: packages/vuetify/src/directives/ripple/index.ts [hunk 3]
```diff
@@ -139,7 +139,7 @@ const ripples = {
           delete el.dataset.previousPosition
         }
 
-        animation.parentNode && el.removeChild(animation.parentNode)
+        if (animation?.parentNode?.parentNode === el) el.removeChild(animation.parentNode)
```
Comment (KaelWD) [inline packages/vuetify/src/directives/ripple/index.ts:142]:
```suggestion
        if (animation.parentNode?.parentNode === el) el.removeChild(animation.parentNode)
```

**Review by KaelWD — APPROVED** [review]

**Pull request merged**